### ISSUE TITLE
Suggestions for KASPA ROI Calculator

### ISSUE THREAD

**Issue by tejas-iot** [coderofstuff/kas-breakeven-calc#1]
Would it be possible to add a feature to calculate the date when the Miner value is zero, past the ROI date?

Also, might have found a error with values, see the file i uploaded.

Appreciate your development efforts!!! 

![kaspa_calc](https://github.com/coderofstuff/kas-breakeven-calc/assets/79765711/49a9490d-3851-4241-b122-070d8282fc58)

**Comment by coderofstuff** [comment]
@tejas-iot if I understood correctly, you're asking about the scenario where you can actually ROI with the device and would just want to see when it would eventually yield 0 net profit after that. That's doable, I'll put it in my todo list.

The Daily Change column showing some big drops in change looks like they happen when Kaspa does its monthly halving.

**Comment by tejas-iot** [comment]
Correct. I’d like to know when a miner crosses to negative cash flow.

Got it on the halving. Thx.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by jakingjack** [comment]
Hello @coderofstuff 
Thanks a lot for your really good tool

some ideas i have ; 
- actually there is an increment for KAS price that we enter ( Daily Price Increase ) ; it would be nice to be able as well to indicate the KAS price at multiple dates ( not increment )
- Get the KAS price prediction by day from https://coindataflow.com/fr/pr%C3%A9diction/kaspa
and be able to chose fo pessimistic value , average or high value
- choose units ( TH/S for hashrate -  PH/s daily increase - ) 
- get automatically the PH/s increase (average year increase ) from https://2miners.com/fr/kas-network-hashrate for example if they have an API
- add an option to choose not to sell each days but in the biggest KAS price days ( colored in the table )
- add pool fees percentage
- add selling platform fees KAS to FIAT
- add USD to EUR fees after KAS to FIAT
- add maintenance fees
- increase session duration stored for the tool ( sometimes it reloads and lose the variables )
- be able to download excel, pdf or txt file of the results

Thanks

**Comment by jakingjack** [comment]
> Hello @coderofstuff Thanks a lot for your really good tool
> 
> some ideas i have ;
> 
>     * actually there is an increment for KAS price that we enter ( Daily Price Increase ) ; it would be nice to be able as well to indicate the KAS price at multiple dates ( not increment )
> 
>     * Get the KAS price prediction by day from https://coindataflow.com/fr/pr%C3%A9diction/kaspa
> 
>     * choose units ( TH/S for hashrate -  PH/s daily increase - )
> 
>     * get automatically the PH/s increase (average year increase ) from https://2miners.com/fr/kas-network-hashrate for example if they have an API
> 
>     * add an option to choose not to sell each days but in the biggest KAS price days ( colored in the table )
> 
>     * add pool fees percentage
> 
>     * add selling platform fees KAS to FIAT
> 
>     * add USD to EUR fees after KAS to FIAT
> 
>     * add maintenance fees
> 
>     * increase session duration stored for the tool ( sometimes it reloads and lose the variables )
> 
>     * be able to download excel, pdf or txt file of the results
> 
> 
> Thanks

Maybe later ;

Add photovoltaic possibility ( price of the solar installation - duration in years - battery capacity in KWH )
-> verify if the consumption of the asics are below battery capacity
-> calculate price per kwh ( total price / nb years of the life durability of the installation / nb of days / nb kwh of battery capacity )

Other idea ; get a json list of known asics + their manufacturer price ( + an option for transport and tax fees ) in the market and select them in a table  ; like 5 ks0 ultra ; another line 5 ks5L ; 5 antminer ks5 pro ....etc etc
it would add the price and the hashrate automatically.

The results would be added together or you could separe them clicking on the + button( like detail by line in the table ).
